File: surf-api-bukkit/surf-api-bukkit-api/src/main/java/dev/slne/surf/surfapi/bukkit/api/packet/listener/listener/PacketListenerResult.java
package dev.slne.surf.surfapi.bukkit.api.packet.listener.listener;

public enum PacketListenerResult {
  /**
   * Continue processing the packet.
   */
  CONTINUE,
  /**
   * Stop processing the packet and cancel the event.
   */
  CANCEL;

  public PacketListenerResult combine(PacketListenerResult other) {
    return this == CANCEL || other == CANCEL ? CANCEL : CONTINUE;
  }
}

File: surf-api-bukkit/surf-api-bukkit-api/src/main/kotlin/dev/slne/surf/surfapi/bukkit/api/packet/listener/listener/PacketListenerResult.kt
package dev.slne.surf.surfapi.bukkit.api.packet.listener.listener

enum class PacketListenerResult {
    /**
     * Continue processing the packet.
     */
    CONTINUE,

    /**
     * Stop processing the packet and cancel the event.
     */
    CANCEL;

    fun combine(other: PacketListenerResult): PacketListenerResult {
        return if (this == CANCEL || other == CANCEL) CANCEL else CONTINUE
    }
}
